conftest.py

```diff
@@ -245,31 +249,51 @@ def pytest_runtest_call(item):
                 logger.info("weblog GRPC request")
 
 
+@pytest.hookimpl(optionalhook=True)
+def pytest_json_runtest_metadata(item, call):
+    if call.when != "setup":
+        return {}
+
+    return _collect_item_metadata(item)
+
+
 def pytest_json_modifyreport(json_report):
 
     try:
         logger.debug("Modifying JSON report")
 
         # populate and adjust some data
         for test in json_report["tests"]:
-            test["skip_reason"] = _skip_reasons.get(test["nodeid"])
+            if "metadata" in test:
+                test["skip_reason"] = (
+                    test["metadata"]["skip_reasons"][test["nodeid"]]
+                    if test["nodeid"] in test["metadata"]["skip_reasons"]
+                    else ""
+                )
 
         # add usefull data for reporting
-        json_report["docs"] = _docs
+        json_report["docs"] = {}
         json_report["context"] = context.serialize()
-        json_report["release_versions"] = _release_versions
-        json_report["rfcs"] = _rfcs
-        json_report["coverages"] = _coverages
+        json_report["release_versions"] = {}
+        json_report["rfcs"] = {}
+        json_report["coverages"] = {}
 
         # clean useless and volumetric data
-        del json_report["collectors"]
+        json_report.pop("collectors", None)
 
         for test in json_report["tests"]:
-            for k in ("setup", "call", "teardown", "keywords", "lineno"):
+            if "metadata" in test:
```

Comment:
once, it's done, we can remove test["metadata"], right ? It will help to keep the report as small as possible.

Reply:
> once, it's done, we can remove test["metadata"], right ? It will help to keep the report as small as possible.

we are removing "metadata" bottom, in the line 291